### ISSUE TITLE
[BUG] - Encoding of URI Components that have a / forward slash

### ISSUE THREAD

**Issue by kpconnell** [amzn/selling-partner-api-docs#1973]
I am still trying to find a solution for how to encode and sign requests when the URI element contains a slash.

so for example:
POST
/products/fees/v0/listings/{sku}/feesEstimate

if {sku} = “mysku/withaslash”…

The supplied dotnet swagger client does not work.

what is the second line of the canonical string to sign to send?
What is the actual raw query string passed to the REST call?
I have tried every combination of encoding I can think of without any love.

**Comment by aibirriaga** [comment]
Same issue here, any news?

**Comment by hamduc7** [comment]
Same issue, any fix?

**Comment by sreecodeslayer** [comment]
@kpconnell Did you find a fix for this. I have the same issue with [Listing Items API](https://developer-docs.amazon.com/sp-api/docs/listings-items-api-v2021-08-01-reference#get-listings2021-08-01itemsselleridsku): `GET /listings/2021-08-01/items/{sellerId}/{sku}` 

I tried digging this a bit deeper by looking at the expected parts of the signature.

Eg.
SKU: `UPWM-2840/EN01`
A forward slash when encoded would be
```elixir
URI.encode_www_form("UPWM-2840/EN01")
> "UPWM-2840%2FEN01"
```
When I send the request, amazon complains with a mismatching Canonical String for request to be having the same SKU encoded as 
```
GET
/listings/2021-08-01/items/AXXXXXXXXXXXX/UPWM-2840%252FEN01
```
This is a double encoded forward slash rather than a single.
```elixir
URI.encode_www_form(URI.encode_www_form("UPWM-2840/EN01"))
"UPWM-2840%252FEN01"
```
I am not sure why this is happening, but a google search tells me this could be a bug in Apache (if that's what amazon is using)

```elixir
URI.decode_www_form("%252F")
> "%2F"
URI.decode_www_form("%2F")
> "/"
```

Just to make sure if there is indeed a problem in how amazon is decoding the url parms, I tried to send the request with double encoded sku like so:
```elixir
URI.encode_www_form(URI.encode_www_form("UPWM-2840/EN01"))
> "UPWM-2840%252FEN01"
```
and we get another complaint from amazon for mismatching canonical string
`GET\n/listings/2021-08-01/items/AJIUH3QFG7FFV/UPWM-2840%25252FEN01`
which is a triple encoded SKU
